File: spring-cloud-integrated-demo/user-service/user-service-api/src/main/java/com/github/jccode/springcloud/integrateddemo/user/api/UserAPI.java
package com.github.jccode.springcloud.integrateddemo.user.api;

import com.github.jccode.springbootsample.core.data.rest.RestResult;
import com.github.jccode.springcloud.integrateddemo.user.model.User;
import org.springframework.cloud.netflix.feign.FeignClient;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;


@FeignClient(value = "user-service", fallback = UserAPIFallback.class)
public interface UserAPI {

    @RequestMapping(value = "/{name}", method = RequestMethod.GET)
    RestResult<User> find(@PathVariable("name") String name);
}

File: spring-cloud-integrated-demo/user-service/user-service-api/src/main/scala/com/github/jccode/springcloud/integrateddemo/user/api/UserApi.scala
package com.github.jccode.springcloud.integrateddemo.user.api

import com.github.jccode.springbootsample.core.data.RestResult.RestSuccess
import com.github.jccode.springcloud.integrateddemo.user.model.User
import org.springframework.cloud.netflix.feign.FeignClient
import org.springframework.web.bind.annotation.{PathVariable, RequestMapping, RequestMethod, RequestParam}

@FeignClient(value = "user-service", fallback = classOf[UserApiFallback])
trait UserApi {

  @RequestMapping(value = Array("/users"), method = Array(RequestMethod.GET))
  def findByName(@RequestParam("name") name: String): RestSuccess[User]

  @RequestMapping(value = Array("/users/{id}"), method = Array(RequestMethod.GET))
  def find(@PathVariable("id") id: Integer): RestSuccess[User]

}
